### PR TITLE
feat: Add new profile fields

### PR DESCRIPTION
**Pull request by pdthompson5**
* Add the following new profile fields:
  * organizations: string[]
  * position: string
  * textBio: string 
  * scirenRegion: string 
* Change joinDate from Date to string
  * Makes handling it quite a bit easier
* Make academicInterests display as "Subjects Taught" if user is a teacher else "Research Areas"
* Clean up some TODOs
  * Make the type system with fetching from the database a bit easier 
* Change user collection to "usersv2"
  *  Most likely temporary
 Database schema change in google drive on usersv2 page

### CODE REVIEW

**Comment by s-kirby** [comment]
Sort grade ranges before pushing user to DB. Also (minor) darken selected color for `formik-mui` grade range / organization.

File: src/pages/edit-profile.tsx [hunk 1]
```diff
@@ -33,9 +33,31 @@ export const gradeRangeOptions = [
...
+  "George Mason University",
+];
+
+export const regionOptions = [
+  "Alabama", 
+  "Coast",
+  "Triangle",
+  "Georgia",
+  "George Mason",
+  "San Diego"
```
Comment (s-kirby) [inline src/pages/edit-profile.tsx:58]: Maybe add "Not Affiliated", "No Region", "Independent" or "None".

**Comment by s-kirby** [comment]
Submission fails without a SciREN organization for both edit user and registrationa

**Comment by pdthompson5** [comment]
> Sort grade ranges before pushing user to DB. Also (minor) darken selected color for `formik-mui` grade range / organization.

It seems that making the color darker is possible but challenging so I'll skip this for now.

**Pull request merged**